utils-plot.R

```diff
@@ -11,51 +11,46 @@ test_data_layer <- function(sol_data, stud_data) {
-                     (length(intersect(sol_rows, stud_cols)) >= length(sol_rows) && 
-                      length(intersect(sol_cols, stud_rows)) >= length(sol_cols))
-        } else if (stud_type == "FacetWrap" && (length(sol_cols) == 0 || length(sol_rows) == 0)){
-            stud_facets <- names(stud_facet$params$facets)
-            equal <- length(intersect(sol_cols, stud_facets)) >= length(sol_cols)||
-                     length(intersect(sol_rows, stud_facets)) >= length(sol_rows)
-        }
-        if(length(sol_cols) == 0 || length(sol_rows) == 0){
-            return(list('equal' = equal, 'feedback' = paste0("Did you make a multipanel plot with rows or columns at least divided by (", c(sol_rows, sol_cols), ")?")))
+    
+    if (ignore_facet_type) {
+        equal <- setequal(sol_facet$vars(), stud_facet$vars())
+        return(list('equal' = equal, 'feedback' = ifelse(equal, "", paste0("Did you make a multipanel plot with rows or columns at least divided by (", paste0(sol_facet$vars(), collapse = ", "), ")?")))) 
+    } else {
+        if (sol_type == "FacetWrap") {
+            if (stud_type  == "FacetWrap") {
+                equal <- setequal(sol_facet$vars(), stud_facet$vars())
+                return(list('equal' = equal, 'feedback' = ifelse(equal, "", paste0("Did you make a multipanel plot with rows or columns at least divided by (", paste0(sol_facet$vars(), collapse = ", "), ")?")))) 
+            } else {
+                return(list('equal' = FALSE, `feedback` = "Did you use the facet_wrap layer?"))
+            }
         } else {
-            return(list('equal' = equal, 'feedback' = paste0("Did you make a multipanel plot with rows and columns respectively divided by (", sol_rows, "), (", sol_cols, ") or the other way arround?")))
-        }
-        
-    } else if (sol_type == "FacetWrap"){
-        sol_facets <- names(sol_facet$params$facets)
-        if (stud_type == "FacetGrid") {
-            stud_cols <- names(stud_facet$params$cols)
-            stud_rows <- names(stud_facet$params$rows)
-            equal <- length(intersect(sol_facets, stud_cols)) >= length(sol_facets)||
-                     length(intersect(sol_facets, stud_rows)) >= length(sol_facets)
-        } else if (stud_type == "FacetWrap") {
-            stud_facets <- names(stud_facet$params$facets)
-            equal <- length(intersect(sol_facets, stud_facets)) >= length(sol_facets)
+            if (stud_type  == "FacetGrid") {
+                
+                sol_rows <- names(sol_facet$params$rows)
+                sol_cols <- names(sol_facet$params$cols)
+                
+                stud_rows <- names(stud_facet$params$rows)
+                stud_cols <- names(stud_facet$params$cols)
+                
+                #Cols and rows can be interchanged
+                equal <- (setequal(sol_rows, stud_rows) & setequal(sol_cols, stud_cols)) | (setequal(sol_rows, stud_cols) & setequal(sol_cols, stud_rows))  
```

Comment:
Is there a reason you're using the single `&` and `|` here? Using `&&` and `||` is usually a bit more efficient since they don't evaluate their right-hand side if the left-hand side is `FALSE` or `TRUE` respectively.

Reply:
No reason, i should indeed use `&&` and `||`.